plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsClient.java

```diff
@@ -266,11 +268,11 @@ private Map<String, Optional<String>> getAllTableSheetExpressionMapping()
         return tableSheetMap.buildOrThrow();
     }
 
-    private static Credential getCredentials(SheetsConfig sheetsConfig)
+    private Credential getCredentials(SheetsConfig sheetsConfig)
     {
         if (sheetsConfig.getCredentialsFilePath().isPresent()) {
             try (InputStream in = new FileInputStream(sheetsConfig.getCredentialsFilePath().get())) {
-                return credentialFromStream(in);
+                return credentialFromStream(in, delegatedUserEmail);
```

Comment:
Here we could use `sheetsConfig#getDelegatedUserEmail` and there by all the method can be static.